opensrp-app/src/main/java/org/smartregister/CoreLibrary.java

```diff
@@ -12,20 +22,38 @@
     private static long buildTimeStamp;
 
     private String ecClientFieldsFile = "ec_client_fields.json";
+    private boolean enableP2pLibrary = false;
+    private P2PAuthorizationService p2PAuthorizationService;
+
 
     public static void init(Context context) {
         init(context, null);
     }
 
     public static void init(Context context, SyncConfiguration syncConfiguration) {
         if (instance == null) {
-            instance = new CoreLibrary(context, syncConfiguration);
+            instance = new CoreLibrary(context, syncConfiguration, false, null);
         }
     }
 
     public static void init(Context context, SyncConfiguration syncConfiguration, long buildVersion) {
         if (instance == null) {
-            instance = new CoreLibrary(context, syncConfiguration);
+            instance = new CoreLibrary(context, syncConfiguration, false, null);
+            buildTimeStamp = buildVersion;
```

Comment:
Could you track build Version differently from buildTimeStamp. 

We may want to disable communication if the apps are of different major and/or minor versions versions.

Define it as string so that it can accommodate[ Software versioning ](https://en.wikipedia.org/wiki/Software_versioning) scheme e.g [major].[minor].[maintenance] format

Reply:
- So we can get the `codeVersion` and the `codeName` from the `BuildConfig` directly in the specific host application that will require it.
- It seems that the the `buildVersion` was supposed to mean the `buildTimeStamp`

Reply:
Cool. Whats then the purpose of  ` buildTimeStamp = buildVersion;`

Reply:
👍